Getting `Cannot find module 'typescript'` in CI environments using `npx auto`

**Describe the bug**
When running auto in an CLI (github actions in our example to be precise) environment using `npx`, it fails with `Cannot find module 'typescript'`

<!-- A clear and concise description of what the bug is. -->

**To Reproduce**
Run `npx auto check-pr` (and probably other commands) in an CLI environment, without running `npm i` or related commands first

**Expected behavior**

Running out of the box, without requiring other dependencies to be installed.

<!-- A clear and concise description of what you expected to happen. -->

**Screenshots**

Full stacktrace:
```
npx: installed 228 in 10.695s
Cannot find module 'typescript'
Require stack:
- /home/runner/.npm/_npx/1535/lib/node_modules/auto/node_modules/@auto-it/core/dist/auto.js
- /home/runner/.npm/_npx/1535/lib/node_modules/auto/dist/parse-args.js
- /home/runner/.npm/_npx/1535/lib/node_modules/auto/dist/bin/auto.js
```
running the command `npx auto check-pr`

**Environment information:**

Github actions CI.
using
- actions/setup-node@v2
- actions/checkout@v2


**Additional context**

Seems to be related to the commit https://github.com/intuit/auto/commit/be5c068ff336aa0a020f5b81a69943c3a4c274f8

--- a/packages/core/src/auto.ts
+++ b/packages/core/src/auto.ts
@@ -74,8 +74,12 @@ import { execSync } from "child_process";
 import isBinary from "./utils/is-binary";
 import { gitReset } from "./utils/git-reset";
 
-if (require.resolve("typescript")) {
-  require("ts-node/register/transpile-only");
+try {
+  if (require.resolve("typescript")) {
+    require("ts-node/register/transpile-only");
+  }
+} catch (error) {
+  // User doesn't have TS installed, cannot write TS plugins
 }
 
 const proxyUrl = process.env.https_proxy || process.env.http_proxy;

